### PR TITLE
Move tests, coordinates, and v0.1 meshes

### LINKED ISSUE #66: Reorganize testcases
Currently, some tests from Decapodes2 sit in `test/runtests.jl`, and others sit in `test/Decapodes2/runtests.jl`.

All tests concerning `Decapodes2`:
- Summation Decapodes
- Multiscale arrays
- Coordinates
- diag2dwd
- meshes
- composition

should be in `test/Decapodes2`, with each testset included into `test/Decapodes2/runtests.jl`.

`test/Decapodes2/runtests` should then be included in `test/runtests.jl`.

### PR DESCRIPTION
**Pull request by lukem12345**
Closes #66 

Alongside moving tests as described in #66 , the following changes are made:
- The implementation of `r`, which computes radius of a point, was fixed
- `coordinates.jl` was moved to `src`, with accompanying exports
- Tests for `coordinates.jl` were written

All tests are passing, except the `multiscalearrays.jl` tests. These tests require meshes which are stored in this repo as json files. These tests should be refactored to use the new `loadmesh` feature.

What makes this complicated is that `loadmesh` assumes all meshes are `EmbeddedDeltaSet2D`s, but the file `periodic_mesh.json` encodes an `EmbeddedDeltaDualComplex2D`. One easy resolution should be to convert this mesh to an `EmbeddedDeltaSet2D`, and simply convert back after loading as required. However, converting to an `EmbeddedDeltaSet2D` from an `EmbeddedDeltaDualComplex2D` is not as simple as:
```julia
julia> converted_mesh = EmbeddedDeltaSet2D{Bool, Point3{Float64}}(periodic_mesh); 
```
nor
```julia
julia> converted_mesh = EmbeddedDeltaSet2D{Bool, Point3{Float64}}();                                                                                                                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                                                                                                                                              
julia> copy_parts!(converted_mesh, periodic_mesh);                                                                                                                                                                                                                                                                                                                                            
                                                                                                                                                                                                                                                                                                                                                                                              
julia> converted_mesh = EmbeddedDeltaDualComplex2D{Bool, Float64, Point3{Float64}}(converted_mesh);                                                                                                                                                                                                                                                                                           
```
Note: This last block runs, but converting `converted_mesh` back into the same dual complex we started with is not a simple call to the constructor.

Decapodes v0.1 tests which require these meshes should also now use the new `loadmesh` feature.

### CODE REVIEW

**Comment by lukem12345** [comment]
Another solution could be to overload `loadmesh` with methods that uses the old `parse_json_acset` when given an argument of the singleton type `Rectangle_30x10` or `Torus_30x10`, for the `plot_mesh.json` and `periodic_mesh.json` respectively.

File: test/Decapodes2/diag2dwd.jl [hunk 1]
```diff
@@ -222,26 +222,3 @@ end
     @test nparts(advdiffdp, :Summand) == 2
 end
 
-
-AdvDiff = quote
```
Comment (jpfairbanks) [inline test/Decapodes2/diag2dwd.jl:226]: Why are these getting deleted?

**Comment by lukem12345** [comment]
On my local branch, I have just changed `src/Decapodes2/decapodes.jl` to export the `Icosphere` constructor from `meshes.jl`.
The `multiscalearrays.jl` tests pass when I `include("test/Decapodes2/multiscalearrays.jl")`, but not from `]test`. Investigating now

**Comment by lukem12345** [comment]
I prepended the calls to `compile` with `Decapodes.` and the tests pass from `]test`. I am going to check the scripts now.

**Comment by lukem12345** [comment]
The rectangular and torus meshes used in the older Decapodes examples and docs are now managed with the `Artifacts` system.

Note that tests pass (including `test/multiscalearrays.jl`, but only with Catlab v0.14.8. Pull request #74 addresses this, and this branch should rebase onto `main` after that branch is merged.

**Comment by lukem12345** [comment]
Just rebased onto `main` to get the recent updates for Catlab v0.14.9, and did a `push —force-with-lease`.

The GitHub actions tests should pass, and this branch should go on and get merged if so.

**Pull request merged**